x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformContext.java

```diff
@@ -214,6 +217,37 @@ Instant getLastStatePersistenceFailureStartTime() {
         return lastStatePersistenceFailureStartTime;
     }
 
+    void resetStartUpFailureCount() {
+        startUpFailureCount.set(0);
+        lastStartUpFailure.set(null);
+        startUpFailureTime = null;
+    }
+
+    int getStartUpFailureCount() {
```

Comment:
Could you add a unit test to `TransformContextTests` similar to `testFailureCount`?